File: src/main/java/br/com/kronos/backend/aplicacao/instituicao/Escala.java
package br.com.kronos.backend.aplicacao.instituicao;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Escala {
    
    private long id;
    private String nome;
    private LocalDate dataInicialVigencia;
    private LocalDate dataFinalVigencia; 
    private Long idInstituicao;
    private int idTipoAbrangencia;  
    private int idTipoAvaliacao;

	@Builder
	public Escala(long id, String nome, LocalDate dataInicialVigencia, LocalDate dataFinalVigencia, Long idInstituicao, 
                    int idTipoAbrangencia, int idTipoAvaliacao) {

		super();
        this.id = id;
        this.nome = exigirTamanho(nome, "Nome da escala", 1, 40);
        this.dataInicialVigencia = exigirNaoNulo(dataInicialVigencia, "Data de início de vigência");
        this.dataFinalVigencia = dataFinalVigencia;
        this.idInstituicao = exigirNaoNulo(idInstituicao, "Instituição");
        this.idTipoAbrangencia = exigirNaoNulo(idTipoAbrangencia, "Tipo de abrangência");  
        this.idTipoAvaliacao = exigirNaoNulo(idTipoAvaliacao, "Tipo de avaliação");       
	}
}

File: poc-kronos-frontend/src/model/instituicao/Escala.js
class Escala {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.nome = dados ? dados.nome : null;
    this.dataInicioVigencia = dados ? dados.dataInicioVigencia : null;
    this.dataFinalVigencia = dados ? dados.dataFinalVigencia : null;
    this.idInstituicao = dados ? dados.idInstituicao : null;
    this.idTipoAbrangencia = dados ? dados.idTipoAbrangencia : null;
    this.nomeTipoAbrangencia = dados ? dados.nomeTipoAbrangencia : null;
    this.idTipoAvaliacao = dados ? dados.idTipoAvaliacao : null;
    this.nomeTipoAvaliacao = dados ? dados.nomeTipoAvaliacao : null;
    this.idCurso = null;
  }

  static validations(required, maxLength, numeric, dataFimMenor) {
    return {
      nome: { required, maxLength: maxLength(40) },
      dataInicioVigencia: { required },
      dataFinalVigencia: { dataFimMenor: dataFimMenor("dataInicioVigencia") },
      idInstituicao: { required },
      idTipoAbrangencia: { required },
      idTipoAvaliacao: { required }
    };
  }

  static validationMessages() {
    return {
      nome: {
        required: "Nome é obrigatório",
        maxLength: "Nome só aceita no máximo 40 caracteres"
      },
      dataInicioVigencia: {
        required: "Data Inicio Vigência é obrigatório"
      },
      dataFinalVigencia: {
        dataFimMenor:
          "Data Final de Vigência não pode ser menor que a Data Início de Vigência"
      },
      idInstituicao: {
        required: "Instituição é obrigatória"
      },
      idTipoAbrangencia: {
        required: "Tipo de abrangência é obrigatória"
      },
      idTipoAvaliacao: {
        required: "Tipo de avaliação é obrigatória"
      }
    };
  }
}
export default Escala;
